src/Storage/Sha1FileHashingService.php

```diff
@@ -178,9 +178,21 @@ private function getCache()
      */
     private function buildCacheKey($fileID, $fs)
     {
+        $filesystem = $this->getFilesystem($fs);
+
         $fsID = $this->getFilesystemKey($fs);
 
-        return base64_encode(sprintf('%s://%s', $fsID, $fileID));
+        // If we can't get a timestamp for the file then we'll use the current timestamp so it
+        // invalidates quickly
+        try {
+            $timestamp = $filesystem->has($fileID) ?
+                $filesystem->getTimestamp($fileID) :
+                DBDatetime::now()->getTimestamp();
+        } catch (Exception $e) {
+            $timestamp = DBDatetime::now()->getTimestamp();
```

Comment:
Did you check that `DBDatetime::now()->getTimestamp()` and `$filesystem->getTimestamp($fileID)` use exactly the same format?

Reply:
It does.